File: backend/app/schemas/analytics.py
"""Pydantic schemas for analytics endpoints."""

from datetime import datetime

from pydantic import BaseModel


class MomentumItem(BaseModel):
    symbol: str
    name: str
    avg_score: float
    momentum: float
    acceleration: float | None = None
    volume: int


class MomentumResponse(BaseModel):
    movers: list[MomentumItem]


class AnomalyAlertItem(BaseModel):
    id: str
    company_symbol: str | None = None
    alert_type: str
    severity: str
    title: str
    description: str
    metric_value: float | None = None
    baseline_value: float | None = None
    z_score: float | None = None
    triggered_at: datetime


class AnomalyResponse(BaseModel):
    alerts: list[AnomalyAlertItem]


class CrossSourceItem(BaseModel):
    symbol: str
    source_a: str
    source_b: str
    score_a: float
    score_b: float
    divergence: float


class CrossSourceResponse(BaseModel):
    signals: list[CrossSourceItem]


class PredictiveSignalItem(BaseModel):
    id: str
    symbol: str
    signal_type: str
    direction: str
    strength: float
    correlation: float | None = None
    lag_hours: int | None = None
    description: str
    generated_at: datetime


class PredictiveResponse(BaseModel):
    signals: list[PredictiveSignalItem]

File: frontend/src/types/analytics.ts
// Analytics API types

export interface MomentumItem {
  symbol: string;
  name: string;
  avg_score: number;
  momentum: number;
  acceleration: number | null;
  volume: number;
}

export interface MomentumResponse {
  movers: MomentumItem[];
}

export interface AnomalyAlertItem {
  id: string;
  company_symbol: string | null;
  alert_type: string;
  severity: string;
  title: string;
  description: string;
  metric_value: number | null;
  baseline_value: number | null;
  z_score: number | null;
  triggered_at: string;
}

export interface AnomalyResponse {
  alerts: AnomalyAlertItem[];
}

export interface CrossSourceItem {
  symbol: string;
  source_a: string;
  source_b: string;
  score_a: number;
  score_b: number;
  divergence: number;
}

export interface CrossSourceResponse {
  signals: CrossSourceItem[];
}

export interface PredictiveSignalItem {
  id: string;
  symbol: string;
  signal_type: string;
  direction: string;
  strength: number;
  correlation: number | null;
  lag_hours: number | null;
  description: string;
  generated_at: string;
}

export interface PredictiveResponse {
  signals: PredictiveSignalItem[];
}
